ResearchKit/Common/ORKHeadlineLabel.m

```diff
@@ -44,7 +46,11 @@ + (UIFont *)defaultFontInSurveyMode:(BOOL)surveyMode {
     CGFloat fontSize = [[descriptor objectForKey: UIFontDescriptorSizeAttribute] doubleValue] - defaultHeadlineSize + ORKGetMetricForWindow(surveyMode ? ORKScreenMetricFontSizeSurveyHeadline : ORKScreenMetricFontSizeHeadline, nil);
     CGFloat maxFontSize = ORKGetMetricForWindow(surveyMode ? ORKScreenMetricMaxFontSizeSurveyHeadline : ORKScreenMetricMaxFontSizeHeadline, nil);
     
-    return ORKLightFontWithSize(MIN(maxFontSize, fontSize));
+    if ([[[CEVRKTheme sharedTheme] themeName] isEqualToString:kThemeAllOfUs]) {
```

Comment:
So quick-and-dirty-get-it-done-now this works, but if we expand to more themes (or even completely customizable-by-the-client themes) this doesn't scale because we end up with ever-growing if-trees everywhere.

Can we conceptually invert this so the sharedTheme provides the value. E.g.

``` objective-c
/* Some other file after the theme is loaded from the server... */
[[CEVRKTheme sharedTheme] setThemeName:themeName];

/* ... */

/* In this file when we need to get the font... */
return [[CEVRKTheme sharedTheme] headlineLabelFont] ?: ORKLightFontWithSize(MIN(maxFontSize, fontSize));
// If `headlineLabelFont` is not defined by the current theme return nil to get the default.
```

Reply:
I like this idea.

Reply:
I took a stab at this - I think this is doable and slims down the lines of code inside base ResearchKit classes, too.